x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/IdentifierBuilder.java

```diff
@@ -48,10 +53,32 @@ public String visitIndexPattern(List<EsqlBaseParser.IndexPatternContext> ctx) {
         List<String> patterns = new ArrayList<>(ctx.size());
         ctx.forEach(c -> {
             String indexPattern = visitIndexString(c.indexString());
+            validateIndexPattern(indexPattern, c);
             patterns.add(
                 c.clusterString() != null ? c.clusterString().getText() + REMOTE_CLUSTER_INDEX_SEPARATOR + indexPattern : indexPattern
             );
         });
         return Strings.collectionToDelimitedString(patterns, ",");
     }
+
+    private static void validateIndexPattern(String indexPattern, EsqlBaseParser.IndexPatternContext ctx) {
+        String[] indices = indexPattern.replace("*", "").split(",");
+        try {
+            for (String index : indices) {
+                if (index.isBlank()) {
+                    continue;
+                }
+                index = removeExclusion(index.strip());
+                String temp = IndexNameExpressionResolver.resolveDateMathExpression(index);
+                index = temp.equals(index) ? index : removeExclusion(temp);
```

Comment:
Yes, it is for index names like this `FROM *, -<-logstash-{now/D}>`, I found that `fieldcaps` calls `IndexNameExpressionResolver`, `DateMathExpressionResolver`, and they actually can resolve double exclusions in this case, this is mainly to remove the double exclusions from the DateMathExpression in index names. I'll add a comment to it.

Reply:
Here is an example
```
+ curl -u elastic:password -X POST 'localhost:9200/_query?format=txt&pretty' -H 'Content-Type: application/json' '-d
{ 
  "query": "FROM *, <logstash-{now/D}>"
}
'
{
  "error" : {
    "root_cause" : [
      {
        "type" : "parse_exception",
        "reason" : "unit [D] not supported for date math [/D]"
      }
    ],
    "type" : "parse_exception",
    "reason" : "unit [D] not supported for date math [/D]"
  },
  "status" : 400
}
+ curl -u elastic:password -X POST 'localhost:9200/_query?format=txt&pretty' -H 'Content-Type: application/json' '-d
{
  "query": "FROM *, -<-logstash-{now/D}>"
}
'
{
  "error" : {
    "root_cause" : [
      {
        "type" : "parse_exception",
        "reason" : "unit [D] not supported for date math [/D]"
      }
    ],
    "type" : "parse_exception",
    "reason" : "unit [D] not supported for date math [/D]"
  },
  "status" : 400
}
```

Reply:
I now see `MetadataCreateIndexService.validateIndexOrAliasName()` throws on names starting with `-`, so the code works around that -- got it.
Interestingly, `_search` accepts this `-<-...` pattern, it just doesn't match. And it also accepts multiple leading minutes within the pattern, like `-<--...`  (ex: `GET foo*,-<--foo-{now/d}>/_search`), which I think this check now won't. 

It looks more of an issue with `_search`, but noteworthy, nevertheless. But IMO we shouldn't accept one minus only within the date math expression, but reject a second: rejecting the 1st one already seems more correct.

Reply:
The more I think about it the more that I'm leaning toward skipping the index names starting with exclusion here. It seems like that it causes a lot of confusions and potential inconsistencies. The goal of this check in parser is to detect invalid index names earlier before more expensive operations that reach data node, like fieldcaps, but we don't want to overblock things. 

Both @bpintea and @astefan found it(validate index names starting with exclusion) confusing, I take it as a sign to not error out in these cases(when there is index name with wildcard and index name starts with exclusion), unless we know for sure that an index name will be invalid, for example `-<logstash-{now/D}>`, fieldcaps will complain about this index name with DateMath, with and without exclusion, because `{now/D}` is invalid.

Reply:
The checks downstream of ESQL leave some invalid index patterns through, like for `-<-index-{now/d}>`, but not other, like for `-<index-{now/D}>`. Both are invalid. IMO we can fail both "early", which is my understanding about the intention of the PR.
This shouldn't be a breaking change, even if the execution becomes more restrictive.